gcalbot/gcalbot/http.go

```diff
@@ -93,22 +93,49 @@ func (h *HTTPSrv) configHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	keybaseConvName := keybaseConv.Channel.Name
+
+	isAdmin, err := func() (bool, error) {
+		switch keybaseConv.Channel.MembersType {
+		case "team": // make sure the member is an admin or owner
+		default: // authorization is per user so let anything through
+			return true, nil
+		}
+		res, err := h.kbc.ListMembersOfTeam(keybaseConv.Channel.Name)
+		if err != nil {
+			return false, err
+		}
+		adminLike := append(res.Owners, res.Admins...)
+		for _, member := range adminLike {
+			if member.Username == keybaseUsername {
+				return true, nil
+			}
+		}
+		return false, nil
+	}()
+	if err != nil {
+		return
+	} else if !isAdmin {
+		// should only be able to configure notifications if isAdmin
+		h.showConfigError(w)
+		return
+	}
+
 	isPrivate := func() bool {
```

Comment:
sorry i missd this before but these ehlpesr should be made more general to take a channel instead of a message instead of being duplicated here